Emby.Server.Implementations/HttpServer/HttpListenerHost.cs

```diff
@@ -651,36 +642,41 @@ public Task ProcessWebSocketRequest(HttpContext context)
             return _socketListener.ProcessWebSocketRequest(context);
         }
 
-        // this method was left for compatibility with third party clients
-        private static string NormalizeEmbyRoutePath(string path)
+        private static string NormalizeUrlPrefix(string baseUrl)
         {
-            if (path.StartsWith("/", StringComparison.OrdinalIgnoreCase))
+            if (string.IsNullOrWhiteSpace(baseUrl))
             {
-                return "/emby" + path;
+                // If baseUrl is empty, return an empty prefix string
+                return "";
+            }
+            else
+            {
+                if (baseUrl.StartsWith("/"))
+                {
+                    // If baseUrl was configured with a leading slash, just return it as-is
+                    return baseUrl;
+                }
+                else
+                {
+                    // If baseUrl was not configured with a leading slash, append one for consistency
+                    return "/" + baseUrl;
+                }
             }
-
-            return "emby/" + path;
         }
 
-        // this method was left for compatibility with third party clients
-        private static string NormalizeMediaBrowserRoutePath(string path)
+        private static string NormalizeEmbyRoutePath(string baseUrl, string path)
         {
-            if (path.StartsWith("/", StringComparison.OrdinalIgnoreCase))
-            {
-                return "/mediabrowser" + path;
-            }
+            return NormalizeUrlPrefix(baseUrl) + "/emby" + path;
```

Comment:
First, you can make this a non-`static` and utilize `_config.Configuration.BaseUrl` here instead of passing it as a 
parameter.
Second, I see little sense in repeatedly normalizing the very same `baseUrl` each time, can we do it _once_? :)

Reply:
Where would I go about doing this once? Near the top of the method? I tried to change as little as possible.

I didn't realize that's what static meant, I could change it sure.

Reply:
Actually, is that really a good idea? As-is, couldn't the user make the change and it work immediately, instead of having to have the server restarted?

Reply:
>I didn't realize that's what static meant, I could change it sure.

Making this change seems to throw errors?

```
HttpServer/HttpListenerHost.cs(692,20): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPrefix()' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Emby.S
erver.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                     
HttpServer/HttpListenerHost.cs(692,53): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPath(string)' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Em
by.Server.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                 
HttpServer/HttpListenerHost.cs(697,20): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPrefix()' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Emby.S
erver.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                     
HttpServer/HttpListenerHost.cs(697,61): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPath(string)' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Em
by.Server.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                 
HttpServer/HttpListenerHost.cs(702,20): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPrefix()' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Emby.S
erver.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                     
HttpServer/HttpListenerHost.cs(702,43): error CS0120: An object reference is required for the non-static field, method, or property 'HttpListenerHost.NormalizeUrlPath(string)' [/var/home/joshua/Projects/Jellyfin/projects/server/jellyfin/Em
by.Server.Implementations/Emby.Server.Implementations.csproj]                                                                                                                                                                                 
```

I'm not sure I understand what this means.

Reply:
I just made all of these functions non-static, and it seems to work, so this should be OK now.

Reply:
Also via testing, found that these normalization functions are only called at startup (for each API endpoint) and then not again during normal operation, so this is OK without them being called just once.